docs/ble.md

```diff
@@ -25,21 +38,23 @@ treat them like decimals.*
 Web IDL
 -------
 This IDL provides an overview of the interface; see below for documentation of
-specific API functions.
+specific API functions.  Click [here](Notes_on_WebIDL.md) for an
+explanation of zephyr.js' WebIDL conventions.
+
 
 ```javascript
 // require returns a BLE object
 // var ble = require('ble');
 
-[NoInterfaceObject]
+[ReturnFromRequire]
 interface BLE: EventEmitter {
     void disconnect(string address);
     void startAdvertising(string name, string[] uuids, string url);
     void stopAdvertising();
-    void setServices(PrimaryService services[]);
-    PrimaryService PrimaryService(PrimaryServiceInit init);
-    Characteristic Characteristic(CharacteristicInit init);
-    Descriptor Descriptor(DescriptorInit init);
+    void setServices(PrimaryService[] services);
+    PrimaryService newPrimaryService(PrimaryServiceInit init);
```

Comment:
is newXXX the standard in webidl to declare a constructor?

Reply:
...no, I don't think there _is_ a standard WebIDL way to declare a constructor of one object as a method of some other object.

...but given that that's what this was doing, I made up a consistent naming scheme and hoped it would suffice (it gets the idea across; I didn't explain it and you intuited its function!).  As with almost anything I write, I'm not married to it and would welcome(!) better ideas.

Reply:
For Web IDL please always refer to the editors draft - the stable one is very old: https://heycam.github.io/webidl/

`PrimaryService newPrimaryService(PrimaryServiceInit init);` is pretty weird

In old fashioned JS (no class) the constructor is a function which has the same name as the object, ie.

```
function PrimaryService(name) {
  this.name = name;
}
```

So in Web IDL you would just write it as

```
[Constructor(USVString name), Exposed=Window]
interface PrimaryService  {
};
```